internal/nix/search.go

```diff
@@ -0,0 +1,103 @@
+package nix
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/pkg/errors"
+	"github.com/samber/lo"
+	"go.jetpack.io/devbox/internal/debug"
+	"go.jetpack.io/devbox/internal/lock"
+)
+
+var ErrPackageNotFound = errors.New("package not found")
+var ErrPackageNotInstalled = errors.New("package not installed")
+
+func PkgExists(pkg string, lock *lock.File) (bool, error) {
+	return InputFromString(pkg, lock).validateExists()
+}
+
+type Info struct {
+	// attribute key is different in flakes vs legacy so we should only use it
+	// if we know exactly which version we are using
+	attributeKey string
+	PName        string
+	Version      string
+}
+
+func (i *Info) String() string {
+	return fmt.Sprintf("%s-%s", i.PName, i.Version)
+}
+
+func PkgInfo(nixpkgsCommit, pkg string) *Info {
+	exactPackage := fmt.Sprintf("%s#%s", FlakeNixpkgs(nixpkgsCommit), pkg)
+	if nixpkgsCommit == "" {
+		exactPackage = fmt.Sprintf("nixpkgs#%s", pkg)
+	}
+
+	results := search(exactPackage)
+	if len(results) == 0 {
+		return nil
+	}
+	// we should only have one result
+	return lo.Values(results)[0]
+}
+
+func search(url string) map[string]*Info {
+	return searchSystem(url, "")
+}
+
+func parseSearchResults(data []byte) map[string]*Info {
+	var results map[string]map[string]any
+	err := json.Unmarshal(data, &results)
+	if err != nil {
+		panic(err)
```

Comment:
Let's change this `panic` to `error` ?

Reply:
similar to above comment. I agree, but not introduced in this PR. Propagating an error is gonna cause a lot more changes, so I would prefer to push that off to a follow up.